### PR TITLE
Remove Hardcoded GitHub URL

### LINKED ISSUE #2195: Remove hardcoded github url
The Github module contains a hard-coded URL, meaning that it’s pinging GitHub during testing. 
https://api.github.com/rate_limit

This should be dynamically set via the appropriate configuration option instead so that it works with our testing server.

### PR DESCRIPTION
**Pull request by devanshbatra04**
closes #2195

### CODE REVIEW

File: src/core/github-api.js [hunk 1]
```diff
@@ -21,8 +21,11 @@ export function githubRequestHeaders(conf) {
 }
 
 export async function getRateLimit(conf) {
+  const githubRateLimitAPI = conf.githubRateLimitAPI
+    ? conf.githubRateLimitAPI
```
Comment (marcoscaceres) [inline src/core/github-api.js]: Please use `conf.githubAPI` instead. The URL should always be:

```JS
new URL("./rate_limit", githubAPI).href
```

So you will need to rename `ratelimit.json` to `rate_limit`.

Comment (devanshbatra04) [reply src/core/github-api.js]: The githubAPI currently defaults to https://api.github.com/repos/${org}/${repo}
I will make convert it to https://api.github.com/repos/ and make the required changes in all other files.

Comment (marcoscaceres) [reply src/core/github-api.js]: Ok, as long as it works that same, then go for it.

Comment (marcoscaceres) [reply src/core/github-api.js]: @devanshbatra04 ... could you check something for me? when you make a `Request` to Github API, I think the Response's headers actually say how many remaining requests there are (`X-Rate-Limit:` or something)... if that's true, maybe we don't need getRateLimit() at all?

Comment (devanshbatra04) [reply src/core/github-api.js]: Thats right. I wanted to take that path earlier. There's a blocker though. I believe we would need the number of requests before we make the multiple requests required. The rate limit api doesnt count to a decrement in requests.

Comment (marcoscaceres) [reply src/core/github-api.js]: Ok, let's fix this first, as it's blocking us from merging other things as travis keeps falling over. Then we can work out a better strategy...

**Comment by marcoscaceres** [comment]
Closed by  #2200

**Pull request closed without merging**